_pytest/main.py

```diff
@@ -283,6 +291,17 @@ def __init__(self, config):
         self.trace = config.trace.root.get("collection")
         self._norecursepatterns = config.getini("norecursedirs")
         self.startdir = py.path.local()
+
+        rootdir_ini = config.getini('rootdir')
```

Comment:
This doesn't seem the proper place to override the `rootdir`, we should instead check if the option was passed in the `determine_setup` function, which is responsible to automatically discover the `rootdir`:

https://github.com/pytest-dev/pytest/blob/621374679b0cc9be4bbb76589fd94f20ecb9e0ec/_pytest/config.py#L1326-L1349

We should obtain the `rootdir` from the command line arguments in `Config.__init__` and pass it to this function, which should use the passed `rootdir` (if any) or go about discovering it like it does today if not passed on the command-line.

Reply:
> We should obtain the rootdir from the command line arguments in Config.__init__

How can I obtain it?
command options are empty at this moment

`self.rootdir_cmd_arg = self.getoption('--rootdir')`
returns 
`ValueError: no option named '--rootdir'`
and there is a function `pytest_cmdline_parse` which executes after config.

`determine_setup` function executes in `_initini` and (as I think) ini files parsed before comand line. Am I right?

btw, how can I attach file from repo here like you?

Reply:
I will have to think about how to obtain the command line option at that moment, but here's how you can post code snippets in comments on GitHub:

https://help.github.com/articles/creating-a-permanent-link-to-a-code-snippet/

Reply:
https://github.com/pytest-dev/pytest/blob/621374679b0cc9be4bbb76589fd94f20ecb9e0ec/_pytest/config.py#L991
ns.rootdir is working and when I pass to 
https://github.com/pytest-dev/pytest/blob/621374679b0cc9be4bbb76589fd94f20ecb9e0ec/_pytest/config.py#L1326
some code like this:
```
    if rootdir_cmd_arg:
        rootdir_abs_path = py.path.local(rootdir_cmd_arg)
        if not os.path.isdir(str(rootdir_abs_path)):
            raise UsageError("Directory '{}' not found. Check your '--rootdir' option.".format(rootdir_abs_path))
        rootdir = rootdir_abs_path
    return rootdir, inifile, inicfg or {}
```

my rootdir was changed, but it's not working =) It try to find modules not in rootdir. Searching solution...